### PR TITLE
Pass sku value in SailThru purchase call.

### PR DESCRIPTION
**Pull request by adeelkhan**
LEARNER-1193

WIP - Passing sku value to Sailthru purchase call.

### CODE REVIEW

File: ecommerce_worker/sailthru/v1/tasks.py [hunk 1]
```diff
@@ -46,6 +46,8 @@ def _build_purchase_item(course_id, course_url, cost_in_cents, mode, course_data
 
     # add vars to item
     item['vars'] = dict(course_data.get('vars', {}), mode=mode, course_run_id=course_id)
+    if course_data.get('sku', None):
```
Comment (ahsan-ul-haq) [inline ecommerce_worker/sailthru/v1/tasks.py]: No need to put None it will be by default

File: ecommerce_worker/sailthru/v1/tasks.py [hunk 2]
```diff
@@ -121,6 +123,16 @@ def _get_course_content(course_id, course_url, sailthru_client, site_code, confi
     return response
 
 
+def _get_verified_course_sku(api_response):
```
Comment (ahsan-ul-haq) [inline ecommerce_worker/sailthru/v1/tasks.py]: Add docstring

File: ecommerce_worker/sailthru/v1/tasks.py (hunk 1 above)
Comment (clintonb) [reply ecommerce_worker/sailthru/v1/tasks.py]: Why is the if statement even necessary? Can the API accept `None` here?

File: ecommerce_worker/sailthru/v1/tasks.py [hunk 3]
```diff
@@ -121,6 +123,16 @@ def _get_course_content(course_id, course_url, sailthru_client, site_code, confi
...
+    if api_response.get('type') == 'verified':
+        for product in api_response.get('products'):
+            for attr_type in product.get('attribute_values'):
+                if attr_type.get('name') == 'certificate_type' and \
+                   attr_type.get('value') == 'verified':
+                    sku = product['stockrecords'][0]['partner_sku']
```
Comment (clintonb) [inline ecommerce_worker/sailthru/v1/tasks.py]: You want to return the SKU with USD currency. Don't assume that is the first one.

**Review by clintonb — APPROVED** [review]

**Review by schenedx — CHANGES_REQUESTED** [review]
We should get the SKU from ecommerce IDA when the function here is called.
Then put the SKU into a new `var` key, together with the SKUs we get from content library

File: ecommerce_worker/sailthru/v1/tasks.py [hunk 4]
```diff
@@ -46,6 +46,8 @@ def _build_purchase_item(course_id, course_url, cost_in_cents, mode, course_data
 
     # add vars to item
     item['vars'] = dict(course_data.get('vars', {}), mode=mode, course_run_id=course_id)
+    if course_data.get('sku'):
```
Comment (schenedx) [inline ecommerce_worker/sailthru/v1/tasks.py]: You should no longer get the `purchase_sku` from course data. Instead, the purchase_sku should be the value passed in from the ecommerce service.

File: ecommerce_worker/sailthru/v1/tasks.py [hunk 5]
```diff
@@ -46,6 +46,8 @@ def _build_purchase_item(course_id, course_url, cost_in_cents, mode, course_data
...
+        item['vars']['sku'] = course_data['sku']
```
Comment (schenedx) [inline ecommerce_worker/sailthru/v1/tasks.py]: You should assign the `purchase_sku` into a new var in `item` dictionary. For example:
`item['vars']['purchase_sku'] = purchase_sku`

File: ecommerce_worker/sailthru/v1/tasks.py [hunk 6]
```diff
@@ -81,7 +83,7 @@ def _record_purchase(sailthru_client, email, item, purchase_incomplete, message_
     return True
 
 
-def _get_course_content(course_id, course_url, sailthru_client, site_code, config):
+def _get_course_content(course_id, course_url, sailthru_client, site_code, config, sku=None):
```
Comment (schenedx) [inline ecommerce_worker/sailthru/v1/tasks.py]: Don't need to pass the sku from ecommerce into this function.

File: ecommerce_worker/sailthru/v1/tasks.py [hunk 7]
```diff
@@ -114,14 +117,14 @@ def _get_course_content(course_id, course_url, sailthru_client, site_code, confi
         if not response:
             logger.error('Could not get course data from Sailthru on enroll/purchase event. '
                          'Calling Ecommerce Course API to get course info for enrollment confirmation email')
-            response = _get_course_content_from_ecommerce(course_id, site_code=site_code)
+            response = _get_course_content_from_ecommerce(course_id, site_code=site_code, sku=sku)
```
Comment (schenedx) [inline ecommerce_worker/sailthru/v1/tasks.py]: Don't need to pass `sku` into `_get_course_content_from_ecommerce`

File: ecommerce_worker/sailthru/v1/tasks.py [hunk 8]
```diff
@@ -114,14 +117,14 @@ def _get_course_content(course_id, course_url, sailthru_client, site_code, confi
...
             if response:
                 cache.set(cache_key, response, config.get('SAILTHRU_CACHE_TTL_SECONDS'))
 
     return response
 
 
-def _get_course_content_from_ecommerce(course_id, site_code=None):
+def _get_course_content_from_ecommerce(course_id, site_code=None, sku=None):
```
Comment (schenedx) [inline ecommerce_worker/sailthru/v1/tasks.py]: You don't need the `sku` passed into this function

File: ecommerce_worker/sailthru/v1/tasks.py [hunk 9]
```diff
@@ -146,7 +149,8 @@ def _get_course_content_from_ecommerce(course_id, site_code=None):
 
     return {
         'title': api_response.get('name'),
-        'verification_deadline': api_response.get('verification_deadline')
+        'verification_deadline': api_response.get('verification_deadline'),
+        'sku': sku
```
Comment (schenedx) [inline ecommerce_worker/sailthru/v1/tasks.py]: `sku` is not needed to return

File: ecommerce_worker/sailthru/v1/tasks.py [hunk 10]
```diff
@@ -286,7 +290,7 @@ def update_course_enrollment(self, email, course_url, purchase_incomplete, mode,
             schedule_retry(self, config)
 
     # Get course data from Sailthru content library or cache
-    course_data = _get_course_content(course_id, course_url, sailthru_client, site_code, config)
+    course_data = _get_course_content(course_id, course_url, sailthru_client, site_code, config, sku)
```
Comment (schenedx) [inline ecommerce_worker/sailthru/v1/tasks.py]: No need to pass the `sku` into the function, and then pass it back into `course_data`

**Review by schenedx — CHANGES_REQUESTED** [review]
Please also update the unit test accordingly

**Comment by adeelkhan** [comment]
@clintonb @schenedx Made requested changes please review.

**Review by clintonb — APPROVED** [review]

**Review by schenedx — APPROVED** [review]
👍

**Pull request merged**